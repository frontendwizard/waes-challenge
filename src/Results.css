.results {
  padding: 32px;
  list-style-type: none;
}

.results > * + * {
  padding-top: 8px;
}
