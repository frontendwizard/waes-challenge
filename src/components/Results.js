import React from "react"
import "./Results.css"

const Results = ({ highlights, color, text }) => (
  <ul className="results">
    {highlights.filter(hg => hg.color === color).map(hg => (
      <li key={hg.id}>
        <span className={hg.color} data-testid="highlight-result">
          {text.slice(hg.range.start, hg.range.end)}
        </span>
      </li>
    ))}
  </ul>
)

export default Results
