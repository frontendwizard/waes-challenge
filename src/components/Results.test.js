import React from "react"
import { render } from "react-testing-library"
import Results from "./Results"

describe("Results", () => {
  it("should match snapshot", () => {
    const { asFragment, unmount } = render(
      <Results
        highlights={[
          { range: { start: 0, end: 4 }, color: "red", id: 0 },
          { range: { start: 5, end: 7 }, color: "red", id: 1 },
        ]}
        color="red"
        text="this is a test"
      />,
    )

    expect(asFragment()).toMatchSnapshot()
    unmount()
  })
  it("should show only highlights that matches color", () => {
    const highlights = [
      { range: { start: 0, end: 4 }, color: "red", id: 0 },
      { range: { start: 5, end: 7 }, color: "blue", id: 1 },
    ]
    const { getByTestId, rerender, unmount } = render(
      <Results highlights={highlights} color="red" text="this is a test" />,
    )

    expect(getByTestId("highlight-result").textContent).toBe("this")
    rerender(
      <Results highlights={highlights} color="blue" text="this is a test" />,
    )
    expect(getByTestId("highlight-result").textContent).toBe("is")
    unmount()
  })
})
