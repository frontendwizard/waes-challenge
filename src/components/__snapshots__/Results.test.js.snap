// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Results should match snapshot 1`] = `
<DocumentFragment>
  <ul
    class="results"
  >
    <li>
      <span
        class="red"
        data-testid="highlight-result"
      >
        this
      </span>
    </li>
    <li>
      <span
        class="red"
        data-testid="highlight-result"
      >
        is
      </span>
    </li>
  </ul>
</DocumentFragment>
`;
